lib/marc_match_key/key.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+### A collection of functions and classes to generate a match key based on the
+###   GoldRush algorithm
+module MarcMatchKey
+  ### Generates the Gov Doc portion of a GoldRush key
+  class Key
+    attr_reader :record
+
+    def initialize(record)
+      @record = record
+    end
+
+    def key
+      [key_part_one, key_part_two, key_part_three].join
+    end
+
+    private
+
+    def key_part_one
+      part = TitleKey.new(record).key
+      part << PubDateKey.new(record).key
+      part << PaginationKey.new(record).key
+      part << EditionKey.new(record).key
+    end
+
+    def key_part_two
+      part = PublisherKey.new(record).key
+      part << TypeKey.new(record).key
+      part << TitlePartKey.new(record).key
+      part << TitleNumberKey.new(record).key
+    end
+
+    def key_part_three
+      part = AuthorKey.new(record).key
+      part << TitleDateKey.new(record).key
+      part << GovDocKey.new(record).key
```

Comment:
Did we want to leave out the GovDocKey, since it tends to be misleading?

Reply:
I have run into problems with using the GovDoc Key for this exact reason, but I think we can add this to the discussion that will be the next step. GoldRush has several things that don't really make sense.